dev_guide/builds/build_inputs.adoc

```diff
@@ -0,0 +1,911 @@
+
+[[using-secrets-s2i-strategy]]
+=== Source-to-Image Strategy
+
+When using a `Source` strategy, all defined input secrets are copied to their
+respective `destinationDir`. If you left `destinationDir` empty, then the
+secrets are placed in the working directory of the builder image.
+
+The same rule is used when a `destinationDir` is a relative path; the secrets
+are placed in the paths that are relative to the image's working directory. The
+`destinationDir` must exist or an error will occur. No directory paths are
+created during the copy process.
+
+[NOTE]
+====
+Currently, any files with these secrets are world-writable (have `0666`
+permissions) and will be truncated to size zero after executing the *_assemble_*
+script. This means that the secret files will exist in the resulting image, but
+they will be empty for security reasons.
+====
+
+[[using-secrets-docker-strategy]]
+=== Docker Strategy
+
+When using a `Docker` strategy, you can add all defined input secrets into your
+container image using the
+link:https://docs.docker.com/engine/reference/builder/#add[`ADD`] and
+link:https://docs.docker.com/engine/reference/builder/#copy[`COPY` instructions] in
+your *_Dockerfile_*.
+
+If you do not specify the `destinationDir` for a secret, then the files will be
+copied into the same directory in which the *_Dockerfile_* is located. If you
+specify a relative path as `destinationDir`, then the secrets will be copied
+into that directory, relative to your *_Dockerfile_* location. This makes the
+secret files available to the Docker build operation as part of the context
+directory used during the build.
+
+[NOTE]
+====
+Users should always remove their input secrets from the final application image
+so that the secrets are not present in the container running from that image.
+However, the secrets will still exist in the image itself in the layer where
+they were added. This removal should be part of the *_Dockerfile_* itself.
+====
+
+[[using-secrets-custom-strategy]]
+=== Custom Strategy
+
+When using a `Custom` strategy, all the defined input secrets are available
+inside the builder container in the *_/var/run/secrets/openshift.io/build_*
+directory. The custom build image is responsible for using these secrets
+appropriately. The `Custom` strategy also allows secrets to be defined as
+described in xref:build_strategies.adoc#custom-secrets[Custom Strategy Options].
+
+There is no technical difference between existing strategy secrets and the input
+secrets. However, your builder image might distinguish between them and use them
+differently, based on your build use case.
+
+The inpout secrets are always mounted into the
+*_/var/run/secrets/openshift.io/build_* directory or your builder can parse the
+`$BUILD` environment variable, which includes the full build object.
+
+[[using-external-artifacts]]
+== Using External Artifacts
```

Comment:
Dropped the "During a Build" part of this section title since it was redundant in context.